src/test/java/bit/javaoop/CompanyTest.java

```diff
@@ -2,47 +2,59 @@
+        //when
+        oldCeo = new CEO("Ala",
+                Arrays.asList(new FixedBudgetManager("A", 1, 1),
+                        new FixedEmployeeNumberManager("B", 2, 2)));
         List<Manager> oldList = oldCeo.getManagersList();
         Company company = new Company("C", oldCeo);
         company.hireNewCeo(newCeo);
+        //then
         assertEquals("Adam", company.getCeo().getName());
         assertEquals(oldList, company.getCeo().getManagersList());
 
 
     }
 
     @Test
-    void tostring() {
-
-        Company company = new Company("CCC");
+    void shouldDoToString() {
+        //given
+        CEO oldCeo = new CEO("Zuza");
+        Company company = new Company("CCC",oldCeo);
         company.hireNewCeo(new CEO("Adam"));
-        FixedBudgetManager fbm1 = new FixedBudgetManager("Adam_budget ", 3000, 900);
-        company.getCeo().hireManager(fbm1);
-        FixedBudgetManager fbm2 = new FixedBudgetManager("Ela_budget ", 30800, 77900);
-        company.getCeo().hireManager(fbm2);
-        FixedEmployeeNumberManager fenm1 = new FixedEmployeeNumberManager("Kasia_limit", 2000, 1);
-        company.getCeo().hireManager(fenm1);
-
+        FixedBudgetManager budgetMangerOne = new FixedBudgetManager("Adam_budget ", 3000, 900);
+        FixedBudgetManager budgetMangerTwo = new FixedBudgetManager("Ela_budget ", 30800, 77900);
+        FixedEmployeeNumberManager limitMangerOne = new FixedEmployeeNumberManager("Kasia_limit", 2000, 1);
         AbstractEmployee employee = new Employee("Garek Majęcki", 25000);
-        fenm1.hireEmployee((Employee) employee);
-        fenm1.hireEmployee(new Employee("Jan", 12));
-        fbm1.hireEmployee(new Employee("Sara", 123123));
-        fbm2.hireEmployee(new Employee("Alan", 11122));
-
 
-        assertEquals(company.getName() + "\n\t" + company.getCeo().getName() + " - CEO\n\t\t" + fbm1.getName() + " - Manager\n\t\t" + fbm2.getName() + " - Manager\n\t\t\tAlan - Employee\n\t\t" + fenm1.getName() + " - Manager\n\t\t\t" + employee.getName() + " - Employee\n", company.toString());
+        //when
+        company.getCeo().hireManager(budgetMangerOne);
+        company.getCeo().hireManager(budgetMangerTwo);
+        company.getCeo().hireManager(limitMangerOne);
+        limitMangerOne.hireEmployeeIfPossible((Employee) employee);
+        limitMangerOne.hireEmployeeIfPossible(new Employee("Jan", 12));
+        budgetMangerOne.hireEmployeeIfPossible(new Employee("Sara", 123123));
+        budgetMangerTwo.hireEmployeeIfPossible(new Employee("Alan", 11122));
+
+        //then
+        assertEquals(company.getName()
+                + "\n\t" + company.getCeo().getName()
+                + " - CEO\n\t\t" + budgetMangerOne.getName()
+                + " - Manager\n\t\t" + budgetMangerTwo.getName()
+                + " - Manager\n\t\t\tAlan - Employee\n\t\t" + limitMangerOne.getName()
+                + " - Manager\n\t\t\t" + employee.getName()
+                + " - Employee\n", company.toString());
```

Comment:
można by to  wyekstrahować do zmiennej result, która znajdowałaby się w bloku given wtedy nasza asercja byłaby ładniejsza : ` assertEquals(company.getName(), result);`